From 7be09751abb938b9ff33c583e3cc26323f73b7f1 Mon Sep 17 00:00:00 2001
From: Simon Shields <simon@lineageos.org>
Date: Fri, 24 Nov 2017 16:05:19 +1100
Subject: [PATCH 31/67] PhoneWindowManager: add LineageButtons volumekey hook

We can't use a generic DeviceKeyHandler here, because we don't want to
override some of the AOSP logic (silencing incoming calls, screenshots)
and we need to know if the device is interactive or not (for screen-off
music controls)

Change-Id: I485c2f6006c5bbd358ba0cbd32917689826c5c6d
---
 .../com/android/server/policy/PhoneWindowManager.java     | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index 4002747548c..5235f5b054e 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -301,6 +301,8 @@ import java.lang.reflect.Constructor;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.lineageos.internal.buttons.LineageButtons;
+
 import dalvik.system.PathClassLoader;
 
 /**
@@ -830,6 +832,7 @@ public class PhoneWindowManager implements WindowManagerPolicy {
     private boolean mAodShowing;
 
     private final List<DeviceKeyHandler> mDeviceKeyHandlers = new ArrayList<>();
+    private LineageButtons mLineageButtons;
 
     private static final int MSG_ENABLE_POINTER_LOCATION = 1;
     private static final int MSG_DISABLE_POINTER_LOCATION = 2;
@@ -6289,6 +6292,10 @@ public class PhoneWindowManager implements WindowManagerPolicy {
                         break;
                     }
                 }
+
+                if (mLineageButtons.handleVolumeKey(event, interactive))
+                    break;
+
                 if (mUseTvRouting || mHandleVolumeKeysInWM) {
                     // Defer special key handlings to
                     // {@link interceptKeyBeforeDispatching()}.
@@ -7731,6 +7738,7 @@ public class PhoneWindowManager implements WindowManagerPolicy {
             }
         }
 
+        mLineageButtons = new LineageButtons(mContext);
         mSystemGestures.systemReady();
         mImmersiveModeConfirmation.systemReady();
 
-- 
2.17.1

